source/validate_instruction.cpp

```diff
@@ -163,10 +163,15 @@ ExtensionSet RequiredExtensions(const ValidationState_t& state,
   if (state.grammar().lookupOperand(type, operand, &operand_desc) ==
       SPV_SUCCESS) {
     assert(operand_desc);
+    // If this operand is incorporated into core SPIR-V before or in the current
+    // target environment, we don't require extensions anymore.
+    if (static_cast<uint32_t>(state.context()->target_env) >=
```

Comment:
Should use spvVersionForTargetEnv here.
Don't rely on enum orderings.

Reply:
Very good catch! I've #1437 to solve it.

Reply:
Done